api_fhir/converters/patientConverter.py

```diff
@@ -0,0 +1,256 @@
+from insuree.models import Insuree, Gender
+from rest_framework.exceptions import NotAuthenticated
+
+from api_fhir.constant import IDENTIFIER_TYPE_SYSTEM, IDENTIFIER_TYPE_ACCESSION_ID_CODE, \
+    IDENTIFIER_TYPE_SOCIAL_BENEFICIARY_ID_CODE, IDENTIFIER_TYPE_PASSPORT_NUMBER_CODE, ISO_DATETIME_FORMAT, \
+    MARIAL_STATUS_UNKNOWN, MARIAL_STATUS_SYSTEM, MARIAL_STATUS_MARRIED, MARIAL_STATUS_NEVER_MARRIED, \
+    MARIAL_STATUS_DIVORCED, MARIAL_STATUS_WIDOWED, ISO_DATE_FORMAT
+from api_fhir.converters import BaseFHIRConverter
+from api_fhir.models import Patient, HumanName, Identifier, IdentifierUse, NameUse, \
+    AdministrativeGender, ContactPointSystem, ContactPointUse
+import core
+
+from api_fhir.models.address import AddressUse, AddressType
+
+
+class PatientConverter(BaseFHIRConverter):
+    @classmethod
+    def to_fhir_obj(cls, imis_insuree):
+        fhir_patient = Patient()
+        cls.build_human_names(fhir_patient, imis_insuree)
+        cls.build_fhir_identifiers(fhir_patient, imis_insuree)
+        cls.build_fhir_birth_date(fhir_patient, imis_insuree)
+        cls.build_fhir_gender(fhir_patient, imis_insuree)
+        cls.build_fhir_marital_status(fhir_patient, imis_insuree)
+        cls.build_fhir_telecom(fhir_patient, imis_insuree)
+        cls.build_fhir_addresses(fhir_patient, imis_insuree)
+        return fhir_patient
+
+    @classmethod
+    def to_imis_obj(cls, fhir_patient, current_user):
+        errors = []
+        imis_insuree = cls.createDefaultInsuree()
+        # TODO the familyid isn't covered because that value is missing in the model
+        # TODO the photoId isn't covered because that value is missing in the model
+        cls.build_imis_names(imis_insuree, fhir_patient, errors)
+        cls.build_imis_identifiers(imis_insuree, fhir_patient)
+        cls.build_imis_birth_date(imis_insuree, fhir_patient, errors)
+        cls.build_imis_gender(imis_insuree, fhir_patient)
+        cls.build_imis_marital(imis_insuree, fhir_patient)
+        cls.build_imis_contacts(imis_insuree, fhir_patient)
+        cls.build_imis_addresses(imis_insuree, fhir_patient)
+
+        if current_user is not None:
+            imis_insuree.audit_user_id = 1  # TODO current_user.id currently is of UUID type not an integer
```

Comment:
I'm not sure what's going on here. Why you assign 1 value ?

Reply:
I mention that in the comment below the ticket. There isn't a simple way to get information about the ID of the currently logged user. I will try to find the best solution for that and then I will create a separate PR for that ticket (I believe it will increase the readability - this PR is quite long).

Reply:
I assigned 1 because of that is ID of the admin user. This approach will be replaced in next PR that's why I didn't create constant for that magic number.